pandas/core/window.py

```diff
@@ -63,19 +63,19 @@ class _Window(PandasObject, SelectionMixin):
         "axis",
         "on",
         "closed",
-    ]
+    ]  # type: List[str]
     exclusions = set()  # type: Set[str]
 
     def __init__(
         self,
         obj,
         window=None,
-        min_periods=None,
-        center=False,
-        win_type=None,
-        axis=0,
-        on=None,
-        closed=None,
+        min_periods: Optional[int] = None,
+        center: Optional[bool] = False,
+        win_type: Optional[str] = None,
+        axis: Union[str, int] = 0,
```

Comment:
Can you put this in pandas._typing as `Axis` and reference here? 

cc @topper-123 as this might conflict with your other PR